ansible/www-standalone/resources/config/nodejs.org

```diff
@@ -174,53 +255,72 @@ server {
         rewrite ^/documentation/api(.*)$ /api$1 redirect;
     }
 
+    # This Location is used for our Metrics Pages
+    # We keep `autoindex` on as some folders are autoindex (do not have an index.html)
+    # even tho the main /metric endpoint is a build page (statically generated over time)
+    # We use ^~ to tell NGINX to not process any other Location directive or Rewrite after this match
     location ^~ /metrics {
         alias /home/dist/metrics;
         autoindex on;
         default_type text/plain;
     }
 
-    location = /github-webhook.log {
-        alias /home/nodejs/github-webhook.log;
-        default_type text/plain;
-    }
-
+    # Store the original language of the request
+    # We'll use this for the 404 in the try_files in @localized_404
     location @english_fallback {
-        # Store the original language of the request
-        # We'll use this for the 404 in the try_files in @localized_404
+        # @TODO: Handle Localization Fallback through Next.js SSR as this is a hacky approach and requires
+        # continuous maintenance of the supported languages
         if ($uri ~* ^/(ar|be|ca|de|es|fa|fr|gl|id|it|ja|ka|ko|nl|pt-br|ro|ru|tr|uk|zh-cn|zh-tw)/) {
             set $lang $1;
         }
 
-        rewrite ^/(ar|be|ca|de|es|fa|fr|gl|id|it|ja|ka|ko|nl|pt-br|ro|ru|tr|uk|zh-cn|zh-tw)/(.*)$ /en/$2 last;
+        rewrite ^/(ar|be|ca|de|es|fa|fr|gl|id|it|ja|ka|ko|nl|pt-br|ro|ru|tr|uk|zh-cn|zh-tw)/(.*)$ /en/$2;
     }
 
+    # This Location directive is used for handling all 404's requests coming to this Server
+    # by default if the request doesn't come from a /two-letter-code/ localized page (extracted from the @english_fallback Location)
+    # it will then fallback to $lang as being "en" as defined numerous lines above
     location @localized_404 {
+        # We disable caching of 404 pages as we want to avoid CloudFlare accidentally thinking that a 404 is permanent
+        # many cases 404 pages happen due to the server reaching maximum concurrent file system open() requests
+        # disabling cache allows CloudFlare to re-evaluate the same $uri once our server recovers and then properly cache the page
         add_header Cache-Control "private, no-store, max-age=0" always;        
 
         # If this was a rewritten i18n request from @english_fallback, use the localized 404
-        # Otherwise, fallback to the English 404 file
+        # if there is no 404 page for that Localized page we will fallback to the English Website 404
+        # as last resort we fallback to NGINX's default 404 on worst scenario. Notice that this result as a [crit] and should never happen
         try_files /$lang/404.html /en/404.html =404;
     }
 
-    location = /traffic-manager {
+    # This Location directive is used to expose our Website GitHub WebHook Logs
+    location = /github-webhook.log {
+        alias /home/nodejs/github-webhook.log;
+        default_type text/plain;
         access_log off;
+    }
 
```

Comment:
Maybe add a no-store cache header here so we can remove the custom rule at the CF level?

![image](https://user-images.githubusercontent.com/2352663/226096841-91d978ae-ffb1-4602-926f-44e74a6d8510.png)

Reply:
Done